sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -176,33 +178,39 @@ namespace Azure { namespace Core {
             // type mismatch
             std::abort();
           }
-          return *reinterpret_cast<const T*>(ptr->Value.get());
+          outputValue = *reinterpret_cast<const T*>(ptr->Value.get());
+          return true;
         }
       }
-      std::abort();
-      // It should be expected that keys may not exist
-      //  That implies we return T* and NOT a T&
+      return false;
     }
 
     /**
-     * @brief Check whether the context has a key matching \p key parameter in it itself, or in the
-     * branch the context belongs to.
+     * @brief Get a value associated with a \p key parameter within this context or the branch of
+     * contexts this context belongs to.
      *
      * @param key A key associated with a context to find.
      *
-     * @return `true` if this context, or the tree branch this context belongs to has a \p key
-     * associated with it. `false` otherwise.
+     * @return A value associated with the context found; an empty value if a specific value can't
```

Comment:
It should've been "A value associated with the *key*  ... if a specific *key* can't be found", but second part of the doc is inaccurate. We won't return an empty value, we would UB, or, practically, `std::abort()`.